src/main/java/com/coveo/spillway/exception/SpillwayLuaResourceErrorException.java

```diff
@@ -0,0 +1,12 @@
+package com.coveo.spillway.exception;
+
+public class SpillwayLuaResourceErrorException extends RuntimeException {
```

Comment:
Error and Exception kinda mean the same thing, I think Error could be removed without loss of meaning

Reply:
👍

Reply:
Also, java.time.Instant#truncatedTo is a bit tricky to adopt in the case of creating bucket timestamp. truncatedTo supports setting defined time period to zero. However, if we set time period of the time periods to like 40 seconds, that means bucket needs to be created such as:

12.00.40Z, 12.01.20Z etc. The existing calculation does it though it looks pretty ugly. For now, we may leave it that way and someone can work on it in future.